csharp/source/autowrap/csharp/dlang.d

```diff
@@ -328,3 +333,33 @@ private string generateMethodErrorHandling(string insideCode, string returnType)
     ret ~= "    }" ~ newline;
     return ret;
 }
+
+private string generateParameter(T)(string name) {
+    import std.datetime : DateTime, Date, TimeOfDay, SysTime, Duration;
+    import std.traits : fullyQualifiedName;
+    import std.traits : ForeachType;
+
+    alias fqn = fullyQualifiedName!(PrimordialType!T);
+    static if (isDateTimeType!T) {
+        return mixin(interp!"fromDatetime!(${fqn})(${name})");
+    } else static if (isDateTimeArrayType!T) {
+        return mixin(interp!"fromDatetime1DArray!(${fqn})(${name})");
+    } else {
+        return name;
+    }
+}
+
+private string generateReturn(T)(string name) {
```

Comment:
Ins't this the same as the function above except for s/from/to/?

Reply:
Yes, it is for now. But in the future it may not be.